src/solc_0.8/common/BaseWithStorage/WithPermitUpgradable.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts v4.4.1 (token/ERC20/extensions/draft-ERC20Permit.sol)
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-0.8/token/ERC20/extensions/draft-IERC20Permit.sol";
```

Comment:
you should use the upgradeable one 
https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/master/contracts/token/ERC20/extensions/draft-IERC20PermitUpgradeable.sol